src/OpenFeature/IFeatureClient.cs

```diff
@@ -4,24 +4,124 @@
 
 namespace OpenFeature
 {
-    internal interface IFeatureClient
+    /// <summary>
+    /// Interface used to resolve flags of varying types.
+    /// </summary>
+    public interface IFeatureClient
     {
+        /// <summary>
+        /// Appends hooks to client
+        /// <para>
+        /// The appending operation will be atomic.
+        /// </para>
+        /// </summary>
+        /// <param name="hooks">A list of Hooks that implement the <see cref="Hook"/> interface</param>
         void AddHooks(IEnumerable<Hook> hooks);
+
+        /// <summary>
+        /// Gets client metadata
+        /// </summary>
+        /// <returns>Client metadata <see cref="ClientMetadata"/></returns>
         ClientMetadata GetMetadata();
 
+        /// <summary>
+        /// Resolves a boolean feature flag
+        /// </summary>
+        /// <param name="flagKey">Feature flag key</param>
+        /// <param name="defaultValue">Default value</param>
+        /// <param name="context"><see cref="EvaluationContext">Evaluation Context</see></param>
+        /// <param name="config"><see cref="EvaluationContext">Flag Evaluation Options</see></param>
+        /// <returns>Resolved flag details <see cref="FlagEvaluationDetails{T}"/></returns>

```

Comment:
This wasn't something you introduced, @cdonnellytx, so I don't mind if you address it or not, but these docs are _slightly_ innacurate. This just resolves the flag _value_ not wrapped in the `FlagEvaluationDetails` wrapper.

Reply:
I can fix it; do you want me to do it in a separate PR or in this one?

Reply:
I'll leave that up to you, I'll review/approve either way.

Reply:
OK, I have updated the docs.